source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -415,6 +434,84 @@ void ProcessorState::continueIfNecessary() {
   }
 }
 
+bool ProcessorState::handleStreamedBodyResponse(const CommonResponse& common_response) {
+  Buffer::OwnedImpl chunk_data;
+  QueuedChunkPtr chunk = dequeueStreamingChunk(chunk_data);
+  ENVOY_BUG(chunk != nullptr, "Bad streamed body callback state");
+  if (common_response.has_body_mutation()) {
+    ENVOY_LOG(debug, "Applying body response to chunk of data. Size = {}", chunk->length);
+    MutationUtils::applyBodyMutations(common_response.body_mutation(), chunk_data);
+  }
+  bool should_continue = chunk->end_stream;
```

Comment:
What if the side stream server doesn't(or JUST forget to) respond with end_stream = true, look like this is not handled and will be stuck there?

Reply:
Yes, if the side stream does not respond or never send end_of_stream to be true. ext_proc filter will keep waiting for for response, and eventually router filter timeout should kick in and destroy the ext_proc filter. This should be same as if backend server misbehaves.

Reply:
I created this issue https://github.com/envoyproxy/envoy/issues/37065 to track the work to add an integration test if the server failed to send response in time and router filter timeout.

Reply:
This theory doesn't sound solid to me and it seems to make all existing individual filter's error handling pointless.

Besides, tightly coupling side stream error with router/upstream will introducing a bad debugging and customer experience, as they are two different errors.

I think we should improve the error handling of this design.

Reply:
Oh, I thought we agreed that no ext_proc timer for FULL_DUPLEX_STREAMED mode. And let the generic Envoy router idle timer(default 15s: ) to take care of the cases if server is not responding.  
And this will make the server be able to buffer more data, and maybe all the way to the end_of_stream received.  Adding an ext_proc specific timer will limit this capability.